drivers/volume/portworx/portworx.go

```diff
@@ -2166,38 +2164,127 @@ func (d *portworx) CollectDiags(n node.Node) error {
 		if err != nil {
 			return fmt.Errorf("failed to collect diags on node %v, Err: %v %v", pxNode.Hostname, err, out)
 		}
+
 		logrus.Debugf("Successfully collected diags on node %v", pxNode.Hostname)
 		return nil
 	}
 
 	url := fmt.Sprintf("http://%s:9014", n.Addresses[0])
 
-	r := &DiagRequestConfig{
-		DockerHost:    "unix:///var/run/docker.sock",
-		OutputFile:    "/var/cores/diags.tar.gz",
-		ContainerName: "",
-		Profile:       false,
-		Live:          true,
-		Upload:        false,
-		All:           true,
-		Force:         true,
-		OnHost:        true,
-		Extra:         false,
-	}
-
 	c, err := client.NewClient(url, "", "")
 	if err != nil {
 		return err
 	}
-	req := c.Post().Resource(pxDiagPath).Body(r)
+	req := c.Post().Resource(pxDiagPath).Body(config)
+
 	resp := req.Do()
 	if resp.Error() != nil {
 		return fmt.Errorf("failed to collect diags on node %v, Err: %v", pxNode.Hostname, resp.Error())
 	}
+
+	if diagOps.Validate {
+		cmd := fmt.Sprintf("test -f %s", config.OutputFile)
+		out, err := d.nodeDriver.RunCommand(n, cmd, opts)
+		if err != nil {
+			return fmt.Errorf("failed to locate diags on node %v, Err: %v %v", pxNode.Hostname, err, out)
+		}
```

Comment:
is this working? I thought you said specifying the output filename isn't working.